### PR TITLE
Ass7

### PR DESCRIPTION
**Pull request by Erdubya**
nightmare pemdas

### CODE REVIEW

File: php-lover/assignment7.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import sys
+
+class Calculator:
+    def __init__(self):
+        self.OPS = {
+            '/': self.div,
+            '*': self.mul,
+            '-': self.sub,
+            '+': self.add
+        }
```
Comment (colinrlly) [inline php-lover/assignment7.py:16]: Yes this is a neato way to do it.

File: php-lover/assignment7.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+
+    def calculate(self, equation: str) -> float:
```
Comment (colinrlly) [inline php-lover/assignment7.py:19]: Chad type defs.

File: php-lover/assignment7.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        l = re.split(r'\s|([\/*+\-\(\)])', equation)
+        eq_arr = [i for i in l if i != '' and i is not None]
+
+        i = 0
+        while i < len(eq_arr):
+            # create the equation queue 
+            x = eq_arr[i]
+            if x == '(':
+                # process parentheses
+                start = i+1
```
Comment (colinrlly) [inline php-lover/assignment7.py:33]: Python has officially published style guidelines and I believe they specify to have spaces around operators.

File: php-lover/assignment7.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    def sub(self, n1: float, n2: float) -> float:
+        return n1 - n2
+
+
+if __name__ == "__main__":
+    c = Calculator()
+    n = len(sys.argv)
+    if n == 2:
+        s = sys.argv[1]
```
Comment (colinrlly) [inline php-lover/assignment7.py:86]: Nice error checking

File: php-lover/assignment7.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    elif n == 1:
+        # example equation
+        s = '4 + 5 * 4 - ( 3 * 3 )'
+    else:
+        raise SyntaxError
+
+    try:
+        r = c.calculate(s)
+    except:
+        print("oopsie woopsie we had a fucky wucky")
```
Comment (colinrlly) [inline php-lover/assignment7.py:96]: Yes very nice error messages

**Review by colinrlly — CHANGES_REQUESTED** [review]
Requested changes.

File: php-lover/assignment7.py (hunk 2 above)
Comment (Erdubya) [reply php-lover/assignment7.py:19]: poggy

File: php-lover/assignment7.py (hunk 1 above)
Comment (Erdubya) [reply php-lover/assignment7.py:16]: poggy

File: php-lover/assignment7.py (hunk 3 above)
Comment (Erdubya) [reply php-lover/assignment7.py]: resolved

File: php-lover/assignment7.py (hunk 4 above)
Comment (Erdubya) [reply php-lover/assignment7.py:86]: poggy

File: php-lover/assignment7.py (hunk 5 above)
Comment (Erdubya) [reply php-lover/assignment7.py:96]: poggy

**Review by colinrlly — APPROVED** [review]
Yes changes good.

**Pull request merged**